### PR TITLE
Feat: JWT Authentication

### PR DESCRIPTION
**Pull request by alexrisch**
Added JWT Authentication

## Summary by CodeRabbit

- **New Features**
	- Introduced constants for cache management (`GC_TIME`, `STALE_TIME`).
	- Added a new hook `useAccountsList` to retrieve account data in the Accounts component.
	- Enhanced authentication flow with new functions for token management.
	- Implemented a deduplication mechanism for fetch requests.
	- Added functionality for signing messages with installation keys.
	- Introduced a custom hook `useAccountsProfiles` for processing account names.
	- Added a new function `getReadableProfile` for retrieving social media data.

- **Bug Fixes**
	- Improved error handling during logout and API interactions.
	- Enhanced error handling in message reply and sending processes.

- **Documentation**
	- Updated comments for clarity on logout tasks and error handling.

- **Chores**
	- Removed unused API signature management functionality.
	- Updated import paths for better organization.
	- Removed obsolete tests related to `getReadableProfile`.

### CODE REVIEW

File: utils/xmtpRN/client.ts [hunk 1]
```diff
@@ -175,3 +175,23 @@ export const requestMessageHistorySyncByAccount = async (account: string) => {
   }
   await requestMessageHistorySync(client);
 };
+
+export type InstallationSignature = {
+  installationPublicKey: string;
+  installationKeySignature: string;
+};
```
[the remark this answers is not in the document]
Comment (technoplato) [reply utils/xmtpRN/client.ts:182]: Code rabbit can you say more about the nonce solution?

Based on our current implementation, what would have to change in the back end and front end to support a nonce and what benefits would that get us?

**Comment by technoplato** [comment]
I can verify that I know this is functional against the new backend changes. 

I've asked for further audit and Slack here: https://xmtp-labs.slack.com/archives/C05GBHEAGPP/p1733926004965479

Specifically, I have questions about what rabbit mentioned about replay attacks, and nonces. 

I think another lingering question we had was about whether or not there were any security applications related to how to install installations with the same inbox ID would create their tokens

File: utils/xmtpRN/client.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (technoplato) [reply utils/xmtpRN/client.ts:182]: OK, would it be safe to say that my main misunderstanding was that we were sending this message: NONCe Payload encrypted when in reality, we are sending it in plain text?

[the remark this answers is not in the document]
Comment (technoplato) [reply utils/xmtpRN/client.ts:182]: Beautiful that definitely helps me understand the process better thanks a bunch of rabbit

**Review by technoplato — APPROVED** [review]

File: utils/api.constants.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export const CONVERSE_ACCESS_TOKEN_KEY = "CONVERSE_ACCESS_TOKEN";
+export const CONVERSE_REFRESH_TOKEN_KEY = "CONVERSE_REFRESH_TOKEN";
+export const XMTP_IDENTITY_KEY = "XMTP_IDENTITY";
+export const XMTP_API_ADDRESS_HEADER_KEY = "xmtp-api-address";
```
[the remark this answers is not in the document]
Comment (alexrisch) [reply utils/api.constants.ts]: Updated!

**Review by thierryskoda — APPROVED** [review]

**Pull request merged**